ddtrace/tracer/textmap.go

```diff
@@ -1433,17 +1433,20 @@ func (*propagatorBaggage) extractTextMap(reader TextMapReader) (*SpanContext, er
 		// Split on the first instance of "=" i.e, `a=b=c` becomes `a: b=c`.
 		key, val, ok := strings.Cut(kv, "=")
 		if !ok {
-			log.Warn("invalid baggage item: %s, dropping", kv)
+			log.Warn("invalid baggage item: %s, dropping header", kv)
+			return &ctx, nil
 		}
 		key = strings.TrimSpace(key)
 		val = strings.TrimSpace(val)
 		if key == "" || val == "" {
-			log.Warn("invalid baggage item: '%s', dropping", kv)
+			log.Warn("invalid baggage item: %s, dropping header", kv)
+			return &ctx, nil
 		}
 		key, errKey := url.QueryUnescape(key)
 		val, errVal := url.QueryUnescape(val)
 		if errKey != nil || errVal != nil {
-			log.Warn("failed to decode baggage item: %s, dropping", kv)
+			log.Warn("invalid baggage item: %s, dropping header", kv)
+			return &ctx, nil
```

Comment:
Let's remove these changes from this PR and rely on https://github.com/DataDog/dd-trace-go/pull/3498 to make these changes. If needed, you can rebase this PR on top of rachel.yang/baggage-malformed-header